genome_preparation.nf

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env nextflow
+
+def helpMessage() {
+    log.info"""
+    Index the provided genome with bowtie and samtools
+    Usage:
+            --fasta                     Fasta file containing the reference genome.
+            --outdir                    Directory where the index will be deployed.
+    """
+}
+
+// Show help message
+if (params.help) exit 0, helpMessage()
+
+params.fasta  = "$baseDir/resources/genome.fa"
+params.outdir = "$baseDir/resources/"
+
+
+basename = file(params.fasta).getName()
+
+/*
+ * Index the provided reference genome using bowtie build
+ */
+process bowtieGenomeIndex {
+
+    publishDir params.outdir,
+        overwrite: false,
+        mode: "move"
+
+    input:
+        path "genome_fasta" from params.fasta
+
+    output:
+        path "$basename.*.bt2"
+
+    """
+    bowtie2-build genome_fasta $basename
+    """
```

Comment:
```suggestion
    input:
        path genome_fasta from params.fasta

    output:
        path "${basename}.*.bt2"

    """
    bowtie2-build ${genome_fasta} ${basename}
    """
```

I think it's better to define `genome_fasta` as a variable than a literal

Reply:
Or, for consistency with other steps, which are all defined relative to "basename", we _could_ change this to

```nextflow
input:
    path "$basename" from params.fasta

[...]

bowtie2-build ${basename} ${basename}
```

But this looks confusing, and we started with something similar.

Reply:
Not sure I agree with this. 
In my head at least, literals are less confusing than variables. 
When I search through the work directory, I prefer when the file names are hard coded rather than dependent on the input file. This said I'm fine either way.

Reply:
I would add that the only reason I've used variable as opposed to literal in some output was to avoid the rather complex renaming functions in the `publishDir` directive. 
I guess all this depends on conventions we just need to agree on.